pkg/controller/core/workload_controller.go

```diff
@@ -915,6 +941,35 @@ func (r *WorkloadReconciler) admittedNotReadyWorkload(wl *kueue.Workload) (strin
 	return "", 0
 }
 
+// handleDRAResources processes DRA resources for the workload and handles specific error conditions
+// with appropriate status updates. Returns true if the workload should be skipped from further processing.
+func (r *WorkloadReconciler) handleDRAResources(ctx context.Context, wl *kueue.Workload, wlCopy *kueue.Workload, cqName string) (bool, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	if !features.Enabled(features.DynamicResourceAllocation) {
+		return false, nil
+	}
+
+	workload.AdjustResources(ctx, r.client, wlCopy)
```

Comment:
Oh, I see the function is called not here, but this is not good. Maybe this is the reason for the build failing?
I would suggest to move it out. If `workload.AdjustResources` and `handleDRAResources` should be called together then wrap them both in a function.